packages/auth/src/AuthProvider/AuthProvider.tsx

```diff
@@ -130,12 +133,24 @@ export function createAuthProvider<
     // Whenever the authImplementation is ready to go, restore auth and reauthenticate
     useEffect(() => {
       async function doRestoreState() {
+        // @MARK: this is where we fetch currentUser from graphql again
+        // because without SSR, initial state doesn't exist
+        // what we want to do here is to conditionally call reauthenticate
```

Comment:
```suggestion
        // because without SSR, initial state doesn't exist.
        // What we want to do here is to conditionally call reauthenticate
```